### ISSUE TITLE
Feature request: add an option to Layout/ExtraSpacing rule for table aligned arguments

### ISSUE THREAD

**Issue by mdesantis** [rubocop/rubocop#5657]
I have this Rails migration:

```ruby
class CreateUsers < ActiveRecord::Migration[5.2]
  def change
    create_table :users do |t|
      t.citext :email,           null: false, index: { unique: true }
      t.citext :username,                     index: { unique: true }
      t.string :password_digest, null: false

      t.timestamps
    end
  end
end
```

`Layout/ExtraSpacing` complains on `t.citext :email,           null: false, index: { unique: true }` line because there's too much space; but I like how they are aligned. I'd love if `Layout/ExtraSpacing` rule had some kind of `AllowTableStyleArgumentsAlignment` option in order to allow this kind of alignment.

## Expected behavior

`t.citext :email,           null: false, index: { unique: true }` line in the example above shouldn't  complain if some kind of `AllowTableStyleArgumentsAlignment` option is enabled

## Actual behavior

`t.citext :email,           null: false, index: { unique: true }` line raises `Layout/ExtraSpacing` rule violation

--------

## RuboCop version

```
> rubocop -V
0.53.0 (using Parser 2.5.0.3, running on ruby 2.4.2 x86_64-linux)
```

**Comment by mvastola** [comment]
Hi, I would just like to second this. This is a major source of annoyance for me and I wish there were some way to have rubocop recognize parameters intentionally aligned this way.

**Comment by esambo** [comment]
Possibly related to #2035

**Comment by hkmaly** [comment]
Reopen HOW?

**Comment by mdesantis** [comment]
To me it's still relevant, maybe not enough traction but still relevant
